src/entry.rs

```diff
@@ -556,7 +556,8 @@ impl<'a> EntryFields<'a> {
                     }
                 }
             }
-            Ok(())
+            f.flush()?;
```

Comment:
I would prefer that this were left out unless there's a comment indicating why it's necessary.

I don't believe this is necessary because the [`flush` implementation on Windows](https://github.com/rust-lang/rust/blob/a9ec99f4201ec33026a468ef1289f98a95b4d71a/src/libstd/sys/windows/fs.rs#L337) is an empty function.

Reply:
It's also empty on unix; I'll move the comment here - it being a no-op today isn't a guarantee it will be a no-op tomorrow, and as file permissions are external state I would expect the type system rust has around files to save us (that would be a rather bigger change), for set_permissions to consume the File and return a new one...

Reply:
Er sorry, so to be clear, I wrote that part of the standard library. The `flush` here is not needed and is distracting from what this function is doing. The `flush` method is simply flushing buffered written data for buffered writers, nothing else. There's no magical buffers for permission bits or things like that, so this needs to be removed.

Reply:
Easy fix then :).